packages/editor/src/store/block-sources/README.md

```diff
@@ -0,0 +1,22 @@
+Block Sources
+=============
+
+By default, the blocks module supports only attributes serialized into a block's comment demarcations, or those sourced from a [standard set of sources](https://developer.wordpress.org/block-editor/developers/block-api/block-attributes/). Since the blocks module is intended to be used in a number of contexts outside the post editor, the implementation of additional context-specific sources must be implemented as an external process.
+
+The post editor supports such additional sources for attributes (e.g. `meta` source).
+
+These sources are implemented here using a uniform interface for applying and responding to block updates to sourced attributes. In the future, this interface may be generalized to allow third-party extensions to either extend the post editor sources or implement their own in custom renderings of a block editor.
+
+## Source API
+
+### `getDependencies`
```

Comment:
Makes me wonder if we could be more granular at some point. Say update only blocks that depend on a specific meta key if that meta change, otherwise don't update any block.

I guess it would complicate things as we need to track blocks per specific key. 

Maybe a future optimization if needed.

Reply:
> Makes me wonder if we could be more granular at some point. Say update only blocks that depend on a specific meta key if that meta change, otherwise don't update any block.

@epiqueras raised a very similar point for block types which depend on properties from a post, since the way those would likely need to be implemented today is to depend on the full post object, since it's only at the point of `apply` that we can determine the relevant property from the schema (since it varies by block type).

I agree we could probably explore this as a future optimization.